sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/PartitionSupervisorImpl.java

```diff
@@ -35,70 +36,59 @@
 
     private RuntimeException resultException;
 
-    private ExecutorService executorService;
+    private Scheduler scheduler;
 
-    public PartitionSupervisorImpl(Lease lease, ChangeFeedObserver observer, PartitionProcessor processor, LeaseRenewer renewer, ExecutorService executorService) {
+    public PartitionSupervisorImpl(Lease lease, ChangeFeedObserver observer, PartitionProcessor processor, LeaseRenewer renewer, Scheduler scheduler) {
         this.lease = lease;
         this.observer = observer;
         this.processor = processor;
         this.renewer = renewer;
-        this.executorService = executorService;
+        this.scheduler = scheduler;
 
-        if (executorService == null) {
-            this.executorService = Executors.newFixedThreadPool(3);
+        if (scheduler == null) {
+            this.scheduler = Schedulers.elastic();
```

Comment:
I suspect if you create the scheduler/executorservice you will be responsible for disposing it as well.
Please make sure we are doing the proper cleanup scheduler/executorservice  on stop/shutdown